src/v2/guide/composition.md

```diff
@@ -0,0 +1,548 @@
+
+<p class="tip">Note that <code>_</code>-prefixed methods are not considered part of the stable API, so they may be changed or removed without warning. Using render helpers is considered relatively safe, but accompanying <a href="unit-testing.html">unit tests</a> are recommended to ensure that everything still works after upgrading Vue to a new version.</p>
+
+### Global Mixins
+
+You can also apply a mixin globally, but do this with caution!
+
+<p class="tip">Global mixins affect <strong>every</strong> Vue instance and component, including third-party components.</p>
+
+One appropriate use case, however, is to inject processing logic for custom options, such as in the [VueFire plugin](https://github.com/vuejs/vuefire#usage):
+
+``` js
+// Inject handler for a `firebase` custom option
+Vue.mixin({
+  created: function () {
+    var firebaseBindings = this.$options.firebase
+    if (firebaseBindings) {
+      // Add instance properties and listeners
+      // for data from firebase.
+    }
+  }
+})
+```
+
+Check out [VueFire's source](https://github.com/search?utf8=%E2%9C%93&q=%22Vue.mixin%22+repo%3Avuejs%2Fvuefire+path%3A%2Fsrc&type=Code) for a complete example.
+
+
+
+## Custom Merge Strategies
+
+Plugins often define custom options, which use the default strategy of overwriting any existing value. If you're the author a plugin where this is undesirable, you can define a **custom merge strategy** for your options with `Vue.config.optionMergeStrategies`.
+
+For example, if your plugin uses a `foo` option:
+
+``` js
+Vue.config.optionMergeStrategies.foo = function (newValue, baseValue) {
+  // Return the correctly merged value
+}
+```
+
+For most object-based options, you can simply use the same strategy used by `methods`:
+
+``` js
+var strategies = Vue.config.optionMergeStrategies
+strategies.foo = strategies.methods
+```
+
+A more advanced example can be found on [Vuex](https://github.com/vuejs/vuex)'s 1.x merging strategy:
+
+``` js
+const merge = Vue.config.optionMergeStrategies.computed
+Vue.config.optionMergeStrategies.vuex = function (newValue, baseValue) {
+  if (!newValue) return baseValue
+  if (!baseValue) return newValue
+  return {
+    getters: merge(newValue.getters, baseValue.getters),
+    state: merge(newValue.state, baseValue.state),
+    actions: merge(newValue.actions, baseValue.actions)
+  }
+}
+```
+
+
+
+## Wrapper Components
+
+As you've seen, advanced composition with extends and mixins can quickly grow complicated. One simpler alternative is creating **wrapper components**, which provide an interface for another component typically used as its root element.
+
+For example, the `ErrorNote` component we [created earlier with extension](#An-Example-Use-Case) could be refactored to:
+
```

Comment:
Should we make mention of Higher Order Components for people coming from a React background? It might be a comparison that would help some people get the idea a bit faster

Reply:
Yeah, will do.